swift/logging/SwiftLogging.h

```diff
@@ -228,4 +224,54 @@ class Logger {
   Log::Level level_;
 };
 
+namespace detail {
+struct SwiftDiagnosticLogWrapper {
+  const SwiftDiagnostic& value;
+};
+
+}  // namespace detail
 }  // namespace codeql
+
+// we intercept this binlog plumbing function providing better overload resolution matches in
+// case the first non-format argument is a diagnostic source, and emit it in that case with the
+// same timestamp
+namespace binlog::detail {
+template <typename Writer, size_t N, typename... T>
+void addEventIgnoreFirst(Writer& writer,
+                         std::uint64_t eventSourceId,
+                         std::uint64_t clock,
+                         const char (&format)[N],
```

Comment:
Nit: Can we make this a `const char*` to reduce the number of instantiations? (We don't use the bound anywhere that I can see.)

Reply:
using `const char*` would make it a worse overload match than the template `Args&&... args` provided by binlog, so the interception would not work. In any case, this code will go as I'm going for one of your suggestions above